.travis.yml

```diff
@@ -128,8 +128,9 @@ after_script:
 
 script:
     - export COVERAGE=`if [ "$TRAVIS_BRANCH" == "master-w-coverage" ]; then echo "--coverage"; else echo ""; fi`
+    - export TESTMON=`if [[ "$TRAVIS_BRANCH" == "master" || "$TRAVIS_BRANCH" == "develop" ]]; then echo ""; else echo "--testmon"; fi`
```

Comment:
I haven't verified myself, but will this work as expected?  We want the full test suite to run on master (and often develop - pending others' input on how often we're going to run this on develop).

Looking at [Travis's environment variables](https://docs.travis-ci.com/user/environment-variables/#Default-Environment-Variables):

> TRAVIS_BRANCH:
> for push builds, or builds not triggered by a pull request, this is the name of the branch.
> for builds triggered by a pull request this is the name of the branch targeted by the pull request.
> for builds triggered by a tag, this is the same as the name of the tag (TRAVIS_TAG).
> 

This makes me think that the if I send a PR to develop, even though my branch is `feature/code_by_dawn`,  `TRAVIS_BRANCH` will be develop, because this is the name of the branch targeted, and the whole test suite would run.

There's also a `TRAVIS_PULL_REQUEST_BRANCH` which looks useful.  [I like this person's approach to getting the current branch name during a Travis PR](https://graysonkoonce.com/getting-the-current-branch-name-during-a-pull-request-in-travis-ci/ )

Reply:
For ember-osf-web, we have Percy tests set to run only on changes to branches on `centerforopenscience`. This way, it catches merge PRs and, if we have anything particularly risky, we can push a branch to `centerforopenscience` to get the Percy tests run on the risky branch. 

For this, I would lean away from running full tests on merges to `develop`, though, because it should be rare that we want to merge to develop and then push immediately to `master`. I think it would make sense to catch merges to `master` and also to set up a cron job for daily tests run in the middle of the night on `develop` (and _maybe_ even `master`, because I'm paranoid). 

If we took all of that together, I think it would be "changes to any branch on `centerforopenscience` except for `develop`, plus a daily test of `develop`". All that being said, we aren't using travis for that particular logic, so I don't have a good example for how to replicate.

The main reason I would avoid merges to `develop` is because hotfixes generate, I think, three sets of tests when they run what with back merges and all, so that's a lot of tests running all at once for a long time. Any other way of only running the full suite once when doing the gitflow hotfix process would be fine as well.